<launch>
    <arg name="serial_no_camera1"    			default="141722070059"/> 			<!-- Note: Replace with actual serial number -->
    <arg name="serial_no_camera2"    			default="134722070628"/> 			<!-- Note: Replace with actual serial number -->
    <arg name="serial_no_camera3"    			default="213622078748"/> 			<!-- Note: Replace with actual serial number -->
    <arg name="camera1"              			default="camera_right"/>		<!-- Note: Replace with camera name -->
    <arg name="camera2"              			default="camera_left"/>		<!-- Note: Replace with camera name -->
    <arg name="camera3"              			default="camera_top"/>		<!-- Note: Replace with camera name -->
    <arg name="tf_prefix_camera1"         default="$(arg camera1)"/>
    <arg name="tf_prefix_camera2"         default="$(arg camera2)"/>
    <arg name="tf_prefix_camera3"         default="$(arg camera3)"/>
    <arg name="initial_reset"             default="false"/>
    <arg name="reconnect_timeout"         default="6.0"/>
    <arg name="align_depth"               default="true"/>
    <arg name="enable_pointcloud"         default="false"/>
    <arg name="enable_sync"               default="true"/>
    <arg name="filters"                   default="pointcloud"/>

    <group ns="$(arg camera1)">
      <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
        <arg name="serial_no"             value="$(arg serial_no_camera1)"/>
        <arg name="tf_prefix"         		value="$(arg tf_prefix_camera1)"/>
        <arg name="initial_reset"         value="$(arg initial_reset)"/>
        <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
        <arg name="align_depth"           value="$(arg align_depth)"/>
        <arg name="enable_pointcloud"     value="$(arg enable_pointcloud)"/>
        <arg name="enable_sync"           value="$(arg enable_sync)"/>
      </include>
    </group>
  
    <group ns="$(arg camera2)">
      <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
        <arg name="serial_no"             value="$(arg serial_no_camera2)"/>
        <arg name="tf_prefix"		          value="$(arg tf_prefix_camera2)"/>
        <arg name="initial_reset"         value="$(arg initial_reset)"/>
        <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
        <arg name="align_depth"           value="$(arg align_depth)"/>
        <arg name="enable_pointcloud"     value="$(arg enable_pointcloud)"/>
        <arg name="enable_sync"           value="$(arg enable_sync)"/>
      </include>
    </group>
  
    <group ns="$(arg camera3)" if="$(eval serial_no_camera3 != '')">
      <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
        <arg name="serial_no"             value="$(arg serial_no_camera3)"/>
        <arg name="tf_prefix"		          value="$(arg tf_prefix_camera3)"/>
        <arg name="initial_reset"         value="$(arg initial_reset)"/>
        <arg name="reconnect_timeout"     value="$(arg reconnect_timeout)"/>
        <arg name="align_depth"           value="$(arg align_depth)"/>
        <arg name="enable_pointcloud"     value="$(arg enable_pointcloud)"/>
        <arg name="enable_sync"           value="$(arg enable_sync)"/>
      </include>
    </group>
  </launch>